### PR TITLE
cli fixes. start windows (cross compile) tests

### PR DESCRIPTION
**Pull request by fundies**

### CODE REVIEW

File: CommandLine/emake/EnigmaCallbacks.cpp [hunk 1]
```diff
@@ -11,24 +11,24 @@ void* CallBack::OutputThread(void*)
...
-
-    delete[] buffer;
+    int pos = 0;
+    while (!_outFile.eof()) {
+      _outFile.seekg(0, _outFile.end);
+      int length = (int)_outFile.tellg() - pos;
+      _outFile.seekg(pos, _outFile.beg);
+ 
+      if (length > 0) {
+		char* buffer = new char[length + 1];
```
Comment (JoshDreamland) [inline CommandLine/emake/EnigmaCallbacks.cpp]: Why did you switch to tabs, here...?

File: .travis.yml [hunk 2]
```diff
@@ -26,6 +27,9 @@ install:
   - chmod +x ./travis.sh
   - make
   - make emake
+  - curl -L https://github.com/fundies/enigma-libs/files/1347135/enigma-libs.zip > enigma-libs.zip
```
Comment (JoshDreamland) [inline .travis.yml]: We should probably rehost these somewhere more stable, at some point.

File: CommandLine/emake/EnigmaCallbacks.cpp (hunk 1 above)
Comment (fundies) [reply CommandLine/emake/EnigmaCallbacks.cpp]: cause I wrote cli w/ spaces 2 years ago and forgot to change my editor back

File: .travis.yml (hunk 2 above)
Comment (fundies) [reply .travis.yml]: Ya. I havent finished getting everything to build. Alure is being a pain. I was going to give you a zip to host  once I build alure and all of gtk's 200 dependancies and etc

**Comment by RobertBColton** [comment]
@JoshDreamland I just want to point out that we should try to be as efficient as we can with the Travis builds to lower their build time and not overtax the CI service, for our own productivity. It seems as though the added dependencies may have slowed this down? I am not sure because Travis is a pretty schizo guy. Either way, allow me to report my observations:

It "Ran for 29 min 10 sec" to complete the tests on your JDI stopgap commit:
https://travis-ci.org/enigma-dev/enigma-dev/builds/281962684
It "Ran for 28 min 1 sec" to complete the tests on my merge of #1082:
https://travis-ci.org/enigma-dev/enigma-dev/builds/281958748
It "Ran for 41 min 13 sec" to complete the tests on this pull request:
https://travis-ci.org/enigma-dev/enigma-dev/builds/282467273

However, it "Ran for 19 min 27 sec" to complete the tests when I requested a rebuild:
https://travis-ci.org/enigma-dev/enigma-dev/builds/282533431

Not sure what caching may have played a role in that difference, but it seems that the average test time has now doubled from these changes. Please understand that this empirical observational evidence may be the result of me experiencing wild hallucinations and may not be conclusive.

**Comment by RobertBColton** [comment]
@JoshDreamland can you also explain to me why 642ba867d5f3d170c857752818278525d65a14f8 forces the shell to be cmd and if we should perhaps undo that? Please review the discussion you had on the forum about this. As far as I understand, @fundies could not get the cross-compiler working in the CLI with it present and removing it seems to have no effect on my end.

**Pull request merged**